services/web-server/src/loaders/auth.js

```diff
@@ -0,0 +1,42 @@
+import DataLoader from 'dataloader';
+import WebServerError from '../utils/WebServerError';
+
+export default (clients, isAuthed, rootUrl, strategies, req, cfg) => {
+  const getCredentials = new DataLoader(queries => {
+    return Promise.all(
+      queries.map(async (provider, accessToken) => {
+        const strategy = strategies[provider];
+
+        if (!strategy) {
+          throw new WebServerError('InputError', `Could not find a strategy for provider ${provider}`);
+        }
+
+        const user = await strategy.userFromRequest(req);
```

Comment:
Passing `req` to this function makes sense if the credentials are attached as headers (as they are for tc-login).  But since they're query parameters (`provider, accessToken`), I think those parameters will need to be passed along, instead.

Reply:
In `mozila-auth0`'s `userFromRequest`, `req` goes through `express-jwt` which I believe this package expects a request object. Once the jwt is verified, `express-jwt` sets the `req.user` property with the user credentials. Would you prefer if we provide both the request and parameters as arguments?

```jsx
const user = await strategy.userFromRequest({
  req,
  parameters: req.body.variables,
});
```

Reply:
I bet that's why the original tc-login implementation expected the JWT in a header instead of in the request body.  That seemed awkward with REST, and seems even more awkward with GraphQL.  I suspect we need to not use `express-jwt`, and instead use a lower-level JWT library to validate and examine the token.  Or we could shift the definition of the GraphQL query to include only the identityProviderId in the query and require that the token be provided in a header for mozilla-auth0 (and maybe for Github too?).